Tests/WitsmlExplorer.Api.Tests/Workers/LogWorkerToolsTests.cs

```diff
@@ -0,0 +1,172 @@
+            Assert.Equal(expectedLog.Uid, log.Uid);
+            Assert.Equal(expectedLog.EndDateTimeIndex, log.EndDateTimeIndex);
+            Assert.Equal(expectedLog.StartDateTimeIndex, log.StartDateTimeIndex);
+            Assert.Equal(expectedLog.LogCurveInfo.Count, log.LogCurveInfo.Count);
+        }
+
+        [Fact]
+        public async Task GetLogDataForCurveTest_OK()
+        {
+            WitsmlLog expectedLog = CreateLog();
+            _witsmlClient.Setup(client =>
+                client.GetFromStoreAsync(It.IsAny<WitsmlLogs>(), It.Is<OptionsIn>((ops) => ops.ReturnElements == ReturnElements.HeaderOnly))).ReturnsAsync(expectedLog.AsItemInWitsmlList);
+            WitsmlLog log = LogUtils.GetSourceLogs(WitsmlLog.WITSML_INDEX_TYPE_MD, 123.11, 123.12, "Depth").Logs.First();
+            LogUtils.SetupGetDepthIndexed(_witsmlClient, (logs) => logs.Logs.First().StartIndex?.Value == "123.11",
+                new() { new() { Data = "123.11,1," }, new() { Data = "123.12,,2" } });
+
+            LogUtils.SetupGetDepthIndexed(_witsmlClient, (logs) => logs.Logs.First().StartIndex?.Value == "123.12",
+                new() { new() { Data = "123.12,,2" } });
+            var sourceLog = await LogWorkerTools.GetLogDataForCurve(_witsmlClient.Object, log, "mnemonic", null);
+            Assert.Equal("123.11,1,", sourceLog.Data[0].Data);
+            Assert.Equal("123.12,,2", sourceLog.Data[1].Data);
+            Assert.Equal("data", sourceLog.Data[0].TypeName);
+            Assert.Equal("data", sourceLog.Data[1].TypeName);
+            Assert.Equal("Depth,DepthBit,DepthHole", sourceLog.MnemonicList);
+            Assert.Equal("data", sourceLog.TypeName);
+            Assert.Equal("m,m,m", sourceLog.UnitList);
+        }
+
+        [Fact]
+        public void CalculateProgressBasedOnIndexTest_OK()
+        {
+            var log = CreateLog();
+            string mnemonicList = "Depth,BPOS";
+            var witmslLogData = GetTestLogData(mnemonicList);
+            var result = LogWorkerTools.CalculateProgressBasedOnIndex(log, witmslLogData);
+            Assert.Equal(0.77777777777777779, result);
+        }
+
+        [Fact]
+        public void GetUpdateLogDataQueriesTest_OK()
+        {
+            string mnemonicList = "Depth,BPOS";
+            var witmslLogData = GetTestLogData(mnemonicList);
+            var batchedQuueries = LogWorkerTools.GetUpdateLogDataQueries("uid",
+                "uidwell", "uidwellbore", witmslLogData, 2, mnemonicList);
+            Assert.Equal(5, batchedQuueries.Count);
+        }
+
+        private static WitsmlLog CreateLog()
+        {
+            return new WitsmlLog
+            {
+                UidWell = WellUid,
+                UidWellbore = WellboreUid,
+                Uid = LogUid,
+                StartDateTimeIndex = "2023-04-19T00:00:03Z",
+                EndDateTimeIndex = "2023-04-19T00:00:12Z",
```

Comment:
Can you change _StartDateTimeIndex_ to "2023-04-19T00:00:00Z" which represents the actual start of data as specified in _GetTestLogData_? 

If you also set _EndDateTimeIndex_ to "2023-04-19T00:00:20Z", the returned data corresponds to 50% of the data (to avoid all those decimals in the assert).

Reply:
yes

Reply:
done